### PR TITLE
Support Python >= 3.6

### PR DESCRIPTION
**Pull request by naoyak**
### Description of changeset
Proposing dropping support for Python versions < 3.6 to reduce maintenance overhead and unlock some dependency upgrades (https://github.com/airbnb/knowledge-repo/pull/300#commitcomment-39416214).

Test Plan:

### CODE REVIEW

**Comment by bulam** [comment]
@naoyak Thanks for putting it in! I think this LGTM. @etr2460 Any thoughts?

**Comment by etr2460** [comment]
@john-bodley or @serenajiang, would you mind taking a look at this?

**Comment by bulam** [comment]
@naoyak I think there might be a lot of users using Python 2.7 on their own computers. If they try to add posts to the repo using the older version of the knowledge repo compatible with Python 2.7 do you think there could be compatability issues?

**Comment by naoyak** [comment]
@bulam personally I think it's reasonable to require Python 3 for those who want to use the latest version of KR, given that Python 2 has been EOL'd and most major libraries have dropped support. Perhaps it makes sense to ship this in a new major/minor release along with your styling revamp in #548?

File: knowledge_repo/app/routes/debug.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def show_views():
 
         methods = ','.join(rule.methods)
         url = url_for(rule.endpoint, **options)
-        line = urllib.unquote("{:50s} {:20s} {}".format(rule.endpoint, methods, url))
+        line = urllib.parse.unquote("{:50s} {:20s} {}".format(rule.endpoint, methods, url))
```
Comment (john-bodley) [inline knowledge_repo/app/routes/debug.py:80]: Maybe use the logic below in terms of importing this.

**Comment by john-bodley** [comment]
@bulam I agree with @naoyak here. It's pretty easy for users to support multiple versions of Python locally via [`pyenv`](https://github.com/pyenv/pyenv).

File: knowledge_repo/app/routes/debug.py (hunk 1 above)
Comment (naoyak) [reply knowledge_repo/app/routes/debug.py:80]: @john-bodley sorry can you clarify? Would it be better to `from urllib.parse import unquote` at the top of the file?

Comment (naoyak) [reply knowledge_repo/app/routes/debug.py:80]: Ah I see what you were referring to. Best to keep consistent 👍

**Pull request merged**